python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py

```diff
@@ -0,0 +1,123 @@
+from typing import Optional, Union, List, Type
+
+import tensorflow as tf
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+# TensorFlow model objects cannot be pickled, therefore we use
+# a callable that returns the model, instead of a model object
+# itself.
+
+
+class TensorflowPredictor(Predictor):
+    """A predictor for TensorFlow models.
+
+    Args:
+        model_definition: A callable that returns a TensorFlow Keras model
```

Comment:
What are the tradeoffs with accepting a callable vs. the model itself? Is this something we should do for torch as well?